src/pulse3D/nb_peak_detection.py

```diff
@@ -101,10 +101,15 @@ def noise_based_peak_finding(
     # use peaks to extract waveform segments from which noise data can be extracted - control over this could be given to the user if required
     segment_size = 10
 
-    while peaks[-1] + segment_size > len(waveform):
+    while len(peaks) > 0 and peaks[-1] + segment_size > len(waveform):
```

Comment:
an empty list is a truthy value so you don't need to take the length of the list here
```python
while peaks and peaks[-1] + segment_size > len(waveform):
...
```

Reply:
When using just peaks in the while loop with normal array this works, But because this is an NumPy array .any() has to be added. 

When not using .any() I get this error

```
 while peaks and peaks[-1] + segment_size > len(waveform):
E       ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```